framework/src/modules/chain/processor/block_processor.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { Pipeline } = require('./pipeline');
+
+class BlockProcessor {
+	constructor() {
+		this.init = new Pipeline();
+		this.create = new Pipeline();
+		this.fork = new Pipeline();
```

Comment:
Please add JS comments for every pipeline here. And specially something new concept introduced here. e.g. `fork`  was not there in earlier implementation.